### PR TITLE
Delegate to orjson on the encoding path

### PR DESCRIPTION
**Pull request by eyurtsev**
Using orjson on the encoding path. This will encode well known types like uuid, datetime, date, time etc.

This PR does not handle the decoding path as decoding can only be done properly once we take into account the jsonschema. 

Rolling this out partially seems fine since:

1) Handling the encoding side is probably more important for well known objects -- users can still roll out decoding code using RunnableLambda and pydantic or we can add run time coercion to runnables.
2) The encoding format will NOT change once we incorporate more changes (i.e., we're using a standard representation of the well known types)

### CODE REVIEW

File: langserve/serialization.py [hunk 1]
```diff
@@ -90,6 +93,16 @@ class _LangChainEncoder(json.JSONEncoder):
     def default(self, obj) -> Any:
         if isinstance(obj, BaseModel):
             return obj.dict()
+        elif isinstance(obj, UUID):
```
Comment (jakerachleff) [inline langserve/serialization.py:96]: can we just do this with `orjson`, or does that make the deserialization hard?

Comment (eyurtsev) [reply langserve/serialization.py:96]: orjson doesn't support object hooks by design, so we can't use just orjson, we'll still need to use our own decoder to allow users to hook into the encoder.

Hooking into the encoder seems like a must for a user to be able to add information for type discrimination on unions

Comment (nfcampos) [reply langserve/serialization.py:96]: I think it does support? https://github.com/ijl/orjson#default

Comment (eyurtsev) [reply langserve/serialization.py:96]: Ah you're right -- it looks like encoding path supports, but not decoding path

**Review by jakerachleff — APPROVED** [review]

**Pull request merged**